spec/Section 6 -- Execution.md

```diff
@@ -951,25 +951,30 @@ streamRecord, variableValues, subsequentPayloads):
     - Set {isCompletedIterator} to {true} on {streamRecord}.
     - Return {null}.
   - Let {payload} be an unordered map.
-  - Let {item} be the item retrieved from {iterator}.
-  - Let {data} be the result of calling {CompleteValue(innerType, fields, item,
-    variableValues, itemPath, subsequentPayloads, parentRecord)}.
-  - Append any encountered field errors to {errors}.
-  - Increment {index}.
-  - Call {ExecuteStreamField(label, iterator, index, fields, innerType, path,
-    streamRecord, variableValues, subsequentPayloads)}.
-  - If {parentRecord} is defined:
-    - Wait for the result of {dataExecution} on {parentRecord}.
-  - If {errors} is not empty:
-    - Add an entry to {payload} named `errors` with the value {errors}.
-  - If a field error was raised, causing a {null} to be propagated to {data},
-    and {innerType} is a Non-Nullable type:
+  - If an item is not retrieved because of an error:
+    - Append the encountered error to {errors}.
     - Add an entry to {payload} named `items` with the value {null}.
+    - Add an entry to {payload} named `errors` with the value {errors}.
```

Comment:
Can we move these two lines down so it's not repeated in both conditions?

Reply:
Hmm, lower down, we have conditions on adding those entries, but when the iterator errors, we don't have conditions. There may be a way to consolidate that I haven't thought of, though.

Reply:
Can we delete this line and move this
```
    - If {errors} is not empty:
      - Add an entry to {payload} named `errors` with the value {errors}.
```

to outside of the `Otherwise:` block?